File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-8.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let ans = -1;
        let start = 0;
        let end = nums.length-1;
        let mid = 0;
        while(start<=end){
            mid = Math.trunc((start+end)/2);
            console.log("mid : ",mid, " :",nums[mid]);
            console.log("startindex",start);
            console.log("endIndex : ",end);
            if(nums[mid]>target){
                 if(nums[end]>=target && nums[start]>target && nums[mid] >=nums[end]){
                    start = mid+1;
                 }else{
                    end = mid-1;
                 }
            }else if(nums[mid]<target){
                if(nums[end]<target && nums[mid]<= nums[end]){
                    end = mid-1;
                }else{
                    start = mid+1;
                }
            }else{
                return mid;
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/house-robber/submission-8.java
class Solution {
    public int rob(int[] nums) {
        int length = nums.length;
        if(length == 1) return nums[0];
        if(length == 2) return Math.max(nums[0],nums[1]);
        if(length == 3) return Math.max(nums[0]+nums[2],nums[1]);
        int[]dp = new int[length];
           dp[0] = nums[0];
           dp[1] = Math.max(nums[1],nums[0]);
           dp[2] = Math.max(nums[0]+nums[2],nums[1]);
        for(int i=3;i<length;i++){
           dp[i] = nums[i] + Math.max(dp[i-2],dp[i-3]);
        }
        return Math.max(dp[length-1],dp[length-2]);
    }
}
